testcases/kernel/syscalls/open/open09.c

```diff
@@ -1,109 +1,50 @@
-/*
- *   Copyright (c) International Business Machines  Corp., 2002
- *   Copyright (c) 2013 Wanlong Gao <gaowanlong@cn.fujitsu.com>
- *
- *   This program is free software;  you can redistribute it and/or modify
- *   it under the terms of the GNU General Public License as published by
- *   the Free Software Foundation; either version 2 of the License, or
- *   (at your option) any later version.
- *
- *   This program is distributed in the hope that it will be useful,
- *   but WITHOUT ANY WARRANTY;  without even the implied warranty of
- *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
- *   the GNU General Public License for more details.
- *
- *   You should have received a copy of the GNU General Public License
- *   along with this program;  if not, write to the Free Software
- *   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
- */
+// SPDX-License-Identifier: GPL-2.0-or-later
 
 /*
- * Description:
- *	1. open an O_WRONLY file, test if read failed.
- *	2. open an O_RDONLY file, test if write failed.
+ * Copyright (c) International Business Machines  Corp., 2002
+ * Copyright (c) 2013 Wanlong Gao <gaowanlong@cn.fujitsu.com>
+ * Copyright (C) 2024 SUSE LLC Andrea Manzini <andrea.manzini@suse.com>
  */
 
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <stdio.h>
-#include <fcntl.h>
-#include "test.h"
-
-char *TCID = "open09";
-int TST_TOTAL = 2;
-
-#define PASSED 1
-#define FAILED 0
+/*\
+ * [Description]
+ * - a file opened in read only mode should not be writable.
```

Comment:
Same issue as before: it's better to provide a full description of the test, rather than a list of steps.

In this case it could be
```
This test verifies that a file opened in read only mode can't be writable and it verifies that a file opened in write only mode can't be readable.
```
You can also use `O_RDONLY` or `O_WRONLY` instead of saying `opened in read only mode`. For example, `file opened with O_RDONLY`, etc. In this case it's up to you.